src/common/transformations/src/transformations/transpose_sinking/ts_utils.cpp

```diff
@@ -82,6 +83,8 @@ TransposeInputsInfo GetFirstTransposeInput(const NodePtr& node,
         auto transpose_node = as_type_ptr<ov::op::v1::Transpose>(input_node);
         if (!transpose_node)
             continue;
+        if (static_transpose_input && transpose_node->get_input_partial_shape(0).rank().is_dynamic())
```

Comment:
According to Transpose specification 
we can determine the rank of the 1st input using the 2nd one (input_order) if it's a constant
<img width="498" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/146308f5-7382-4de9-be7e-087fc5fb5d16">

The current problematic case is only when 2nd input is Constant and it's an empty tensor. This is a special value for this input. In this case we can't determine the rank. 
So it looks like we can just check that the 2nd input is empty tensor and the 1st input has dynamic rank.